File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/crash/CrashCaptureKit.java
package com.didichuxing.doraemonkit.kit.crash;

import android.content.Context;

import com.didichuxing.doraemonkit.R;
import com.didichuxing.doraemonkit.config.CrashCaptureConfig;
import com.didichuxing.doraemonkit.constant.FragmentIndex;
import com.didichuxing.doraemonkit.kit.AbstractKit;
import com.didichuxing.doraemonkit.kit.Category;

/**
 * Created by wanglikun on 2019/6/12
 */
public class CrashCaptureKit extends AbstractKit {
    @Override
    public int getCategory() {
        return Category.TOOLS;
    }

    @Override
    public int getName() {
        return R.string.dk_kit_crash;
    }

    @Override
    public int getIcon() {
        return R.drawable.dk_crash_catch;
    }

    @Override
    public void onClick(Context context) {
        startUniversalActivity(context, FragmentIndex.FRAGMENT_CRASH);
    }

    @Override
    public void onAppInit(Context context) {
        CrashCaptureManager.getInstance().init(context);
        if (CrashCaptureConfig.isCrashCaptureOpen()) {
            CrashCaptureManager.getInstance().start();
        } else {
            CrashCaptureManager.getInstance().stop();
        }
    }

    @Override
    public boolean isInnerKit() {
        return true;
    }

    @Override
    public String innerKitId() {
        return "dokit_sdk_comm_ck_crash";
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/crash/CrashCaptureKit.kt
package com.didichuxing.doraemonkit.kit.crash

import android.content.Context
import com.didichuxing.doraemonkit.R
import com.didichuxing.doraemonkit.config.CrashCaptureConfig
import com.didichuxing.doraemonkit.constant.FragmentIndex
import com.didichuxing.doraemonkit.kit.AbstractKit

/**
 * Created by wanglikun on 2019/6/12
 */
class CrashCaptureKit : AbstractKit() {
    override val name: Int
        get() = R.string.dk_kit_crash

    override val icon: Int
        get() = R.mipmap.dk_crash_catch

    override val isInnerKit: Boolean
        get() = true

    override fun onClick(context: Context?) {
        startUniversalActivity(context, FragmentIndex.FRAGMENT_CRASH)
    }

    override fun onAppInit(context: Context?) {
        CrashCaptureManager.instance.init(context)
        if (CrashCaptureConfig.isCrashCaptureOpen) {
            CrashCaptureManager.instance.start()
        } else {
            CrashCaptureManager.instance.stop()
        }
    }

    override fun innerKitId(): String {
        return "dokit_sdk_comm_ck_crash"
    }
}